doc/SONiC_OC_NTP_HLD.md

```diff
@@ -0,0 +1,491 @@
+sonic(config)#ntp server 
+String  NTP server address or name
+
+sonic(config)# ntp server 10.11.0.1
+
+sonic(config)# ntp server 2001:aa:aa::a
+
+sonic(config)# ntp server pool.ntp.org
+
+```
+
+##### 3.6.2.1.2 Delete NTP server
+
+```
+sonic(config)# no ntp server
+  String  NTP server address or name
+
+sonic(config)# no ntp server 10.11.0.1
+
+sonic(config)# no ntp server 2001:aa:aa::a
+
+sonic(config)# no ntp server pool.ntp.org
+
+```
+
+##### 3.6.2.1.3 Configure NTP source ip
+
+```
+sonic(config)# ntp
+    source              Configure source IP address
+
+sonic(config)# ntp source 11.22.33.55
+
+sonic(config)# ntp source 2001:aa:aa::b
+
+```
+
+##### 3.6.2.1.4 Delete NTP source
+
+```
+sonic(config)# no ntp source
+
+```
+
+##### 3.6.2.1.5 Configure NTP vrf
+
+```
+sonic(config)#
+    vrf          Enabling NTP on a VRF
+
+sonic(config)#ntp vrf
+  management                  Enable NTP on management VRF
+  String(Max: 32 characters)  Enable NTP on non-default VRF
+
+sonic(config)# ntp vrf management
+
+```
+
+##### 3.6.2.1.6 Delete NTP vrf
+
+```
+sonic(config)# no ntp
+    vrf  Disable NTP on a VRF
+
+sonic(config)# no ntp vrf
+  management                  Disable NTP on management VRF
+  String(Max: 32 characters)  Disable NTP on non-default VRF
+
+sonic(config)# no ntp vrf management
+
+```
+
+#### 3.6.2.2 Show ntp associations
+
+```
+sonic(config)# do show ntp
+  associations  NTP associations
+
+
+sonic(config)# do show ntp associations
+     remote           refid      st t when poll reach   delay   offset  jitter
+==============================================================================
+*10.11.0.1       10.11.8.1        4 u   28   64    1    0.183    1.499   2.625
++2001:aa:aa::b   60.39.129.68    10 u   27   64    1    0.638  2171.31   0.411
++10.11.0.2       10.11.8.1        4 u   24   64    1    0.240  -13.957  12.786
+* master (synced), # master (unsynced), + selected, - candidate, ~ configured
+
+```
+
+#### 3.6.2.3 Debug Commands
+
+#### 3.6.2.4 IS-CLI Compliance
+
+### 3.6.3 REST API Support
+```
+GET - Get existing NTP configuration information from CONFIG DB.
+      Get NTP peer states 
+POST - Add NTP configuration into CONFIG DB.
+PATCH - Update existing NTP configuraiton information in CONFIG DB.
+DELETE - Delete a existing NTP configuration from CONFIG DB.
+```
+
+# 4 Flow Diagrams
+
+# 5 Error Handling
+
+# 6 Serviceability and Debug
+
+# 7 Warm Boot Support
+
+# 8 Scalability
+
+# 9 Unit Test
+
+The unit-test for this feature will include:
+#### Configuration via CLI
+
+| Test Name | Test Description |
+| :-------- | :----- |
+| Configure NTP server | Verify NTP servers are installed correctly in the configDB and reflected in the NTP peers |
+| Delete NTP server | Verify NTP servers are installed correctly in the configDB and reflected in the NTP peers  |
+| Configure NTP source| Verify NTP source is installed correctly in the configDB, NTP packets are transmitted and received over this source |

```

Comment:
Can there be a UT for NTP source address/ip with Mgmt VRF configured ?

Reply:
Yes.